Reorganized chef-backend.rb settings and add etcd.ionice settings (released in Chef Backend 2.2)

btm opened the pull request:
Move chef-backend.rb settings to the common place
    
    Having the Chef Backend settings deep under Setup -> Chef Infra Server
    -> Install High Availability, but not anywhere under Managing Chef Infra
    Server means I can never find them.
    
    All of the -ctl commands for Infra Server are under 'Managing Chef Infra
    Server' as is the chef-server.rb settings. This moves the the settings
    to the same location and links back. These rarely need to be modified,
    so they really should not be all spelled out at length on that page.
    
    There's a slight reordering of the menu to group the server, backend,
    and expander options in it.

--

    Add etcd.ionice configurations
    
    As of 2.1.13 (to be released in 2.2) you can configure the ionice values
    for etcd.

btm committed:
Move chef-backend.rb settings to the common place

Having the Chef Backend settings deep under Setup -> Chef Infra Server
-> Install High Availability, but not anywhere under Managing Chef Infra
Server means I can never find them.

All of the -ctl commands for Infra Server are under 'Managing Chef Infra
Server' as is the chef-server.rb settings. This moves the the settings
to the same location and links back. These rarely need to be modified,
so they really should not be all spelled out at length on that page.

There's a slight reordering of the menu to group the server, backend,
and expander options in it.

Signed-off-by: Bryan McLellan <btm@loftninjas.org>

btm committed:
Add etcd.ionice configurations

As of 2.1.13 (to be released in 2.2) you can configure the ionice values
for etcd.

Signed-off-by: Bryan McLellan <btm@loftninjas.org>

phiggins commented on themes/docs-new/layouts/shortcodes/config_rb_backend_summary.md:
```diff
@@ -0,0 +1,17 @@
+`chef-backend-ctl gen-sample-backend-config` and controls most of the
+various feature and configuration flags going into a Chef HA backend
+node. A number of these options control the reliability, stability and
+uptime of the backend PostgreSQL databases, the elastic search index,
```
```suggestion
uptime of the backend PostgreSQL databases, the Elasticsearch index,
```

"Elasticsearch" is the name of the product, so this PR should probably have a `s/elastic search/Elasticsearch/g` run on it.

stevendanna reviewed (approved):
Technically the writeup looks great.  Thanks for working on this

btm committed:
Replace 'elastic search' with 'Elasticsearch' everywhere

Signed-off-by: Bryan McLellan <btm@loftninjas.org>

phiggins commented on content/config_rb_backend.md:
```diff
@@ -0,0 +1,290 @@
+    different between elected leader and current node in bytes.
+    `52428800` (50MB) by default.
+-   `leaderl.health_check.max_elasticsearch_failures` Number of Elastic
+    Search API failures allowed before health check fails. 5 by default.
```
```suggestion
-   `leaderl.health_check.max_elasticsearch_failures` Number of
    Elasticsearch API failures allowed before health check fails. 5 by default.
```

btm committed:
Replace all instances of 'Elastic Search' with 'Elasticsearch'

Signed-off-by: Bryan McLellan <btm@loftninjas.org>

btm committed:
Merge branch 'master' into btm/backend

IanMadd commented on themes/docs-new/layouts/shortcodes/config_rb_backend_summary.md:
```diff
@@ -0,0 +1,17 @@
+-   `ip_version` Set to either `'ipv4'` or `'ipv6'`. `'ipv4'` by
+    default.
+-   `publish_address` Externally resolvable IP address of this back-end
+    node.
```
Are these settings that the user could or should change?
I'm thinking prefacing this list with something like:
`Verify that the following settings are correctly configured:`
or
`Configure the following settings:`

btm replied:
I think it is more like "these are the only settings you should change when setting up without guidance" - @sean-horn ?

IanMadd commented on content/config_rb_backend.md:
```diff
@@ -0,0 +1,290 @@
+    for alternative formats. `["samehost", "samenet"]` by default.
+-   `postgresql.replication_user` Username used by postgres streaming
+    replicator when accessing this node. `'replicator'` by default.
+-   `postgresql.username` `'chef_pgsql'` by default.
```
Is the `postgresql.username` the database administrator username? Something like:

`postgresql.username` Database administrator username, `'chef_pgsql'` by default.

IanMadd commented:
Edited pages:
https://deploy-preview-2626--chef-web-docs.netlify.app/config_rb_backend/
https://deploy-preview-2626--chef-web-docs.netlify.app/install_server_ha/#chef-backendrb-options

IanMadd commented on content/config_rb_backend.md:
```diff
@@ -0,0 +1,290 @@
+    milliseconds. This is the frequency at which the leader will send
+    heartbeats to followers. Etcd's documentation recommends that this
+    is set roughly to the round-trip times between members. (The default
+    before 1.2 was 100)
```
"(The default before 1.2 was 100)"
Does this refer to Chef Backend version 1.2?

Commit:
Reformat flag lists into definition lists

Signed-off-by: IanMadd <imaddaus@chef.io>

IanMadd committed:
Merge branch 'master' into btm/backend

IanMadd committed:
Merge branch 'master' into btm/backend

btm committed:
Clarify Chef Backend username settings

`postgresql.username` and `leaderl.username` are system users that the
processes run under.

Signed-off-by: Bryan McLellan <btm@loftninjas.org>
